src/Microsoft.SqlTools.ServiceLayer/Rename/RenameService.cs

```
@@ -0,0 +1,138 @@
+        {
+            return this.HandleRequest<bool>(requestContext, async () =>
+           {
+               Logger.Verbose("Handle Request in ProcessRenameEditRequest()");
+               bool operationExecutedSuccessFull = false;
+               RenameUtils.Validate(requestParams);
+               ConnectionInfo connInfo;
+               try
+               {
+                   connectionService.TryFindConnection(
+                          requestParams.TableInfo.OwnerUri,
+                          out connInfo);
+
+                   using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "RenamingDatabaseObjects"))
+                   {
+                       ExecuteRenaming(requestParams, sqlConn);
+                   }
+                   operationExecutedSuccessFull = true;
+               }
+               catch (Exception e)
+               {
+                   Logger.Error("Error on executing renaming operation: " + e.ToString());
+                   throw new InvalidOperationException("The renaming operation was not successfull executed");
+               }
+               await requestContext.SendResult(operationExecutedSuccessFull);
+           });
+        }
+
+        /// <summary>
+        /// Method to execute the renaming operation on the server
+        /// </summary>
+        /// <param name="requestParams">parameters which are needed to execute renaming operation</param>
+        /// <param name="sqlConn">the sqlconnection on which the operation is executed</param>
+        private void ExecuteRenaming(ProcessRenameEditRequestParams requestParams, SqlConnection sqlConn)
+        {
+            Logger.Verbose("Inside in the ExecuteRenaming()-Method");
+            IRenamable renameObject = RenameUtils.GetSQLRenameObject(requestParams, sqlConn);
+            try
+            {
+                renameObject.Rename(requestParams.ChangeInfo.NewName);
+            }
+            catch (Exception e)
```

Just remove this - we already catch it above and add additional info to the message so we can let this bubble up.